File: frontend/src/di/infrastructures.ts
import { HTTP } from '../adapters/infrastructures/HTTP';
import { TextbookHTTPQueryInfrastructure } from '../adapters/infrastructures/textbook/TextbookHTTPQueryInfrastructure';
import { AnswerHTTPQueryInfrastructure } from '../adapters/infrastructures/answer/AnswerHTTPQueryInfrastructure';
import { HintHTTPQueryInfrastructure } from '../adapters/infrastructures/hint/HintHTTPQueryInfrastructure';
import { ProblemHTTPQueryInfrastructure } from '../adapters/infrastructures/problem/ProblemHTTPQueryInfrastructure';


export const infrastructures = {
  textbookHTTPQueryInfrastructure:  new TextbookHTTPQueryInfrastructure(HTTP),
  answerHTTPQueryInfrastructure: new AnswerHTTPQueryInfrastructure(HTTP),
  hintHTTPQueryInfrastructure:  new HintHTTPQueryInfrastructure(HTTP),
  problemHTTPQueryInfrastructure: new ProblemHTTPQueryInfrastructure(HTTP),
}

File: checker/src/framework/di/adapters/infrastructures.py
from ....adapters import infrastructures

mock_db_command_infrastructure = (infrastructures
                                  .answer
                                  .MockDbCommandInfrastructure()
                                  )

mock_textbook_db_query_infrastructure = (
    infrastructures.textbook.MockDbQueryInfrastructure())
mock_hint_db_query_infrastructure = (
    infrastructures.hint.MockDbQueryInfrastructure())
mock_problem_db_query_infrastructure = (
    infrastructures.problem.MockDbQueryInfrastructure())
